components/cdc/src/endpoint.rs

```diff
@@ -626,11 +646,36 @@ impl<T: 'static + RaftStoreRouter<RocksEngine>> Endpoint<T> {
                             region_id,
                             err: Error::Request(e.into()),
                         };
+                        tx.clone().send(Either::Right(region_id)).wait().unwrap();
                         if let Err(e) = scheduler.schedule(Task::Deregister(deregister)) {
                             error!("schedule cdc task failed"; "error" => ?e);
                         }
                     }
                 }
+
+                let mut ready_regions = vec![];
+                let mut ready_count = 0;
+                let wait_tso_fut = rx.for_each(move |res| {
+                    if let Either::Left(region_id) = res {
+                        ready_regions.push(region_id);
+                    }
+                    ready_count += 1;
+                    if ready_count >= region_count {
+                        match scheduler_clone.schedule(Task::MinTS {
+                            min_ts,
+                            regions: ready_regions.clone(),
+                        }) {
+                            Ok(_) | Err(ScheduleError::Stopped(_)) => (),
+                            Err(err) => panic!(
+                                "failed to schedule min_ts event, min_ts: {}, error: {:?}",
+                                min_ts, err
+                            ),
+                        }
+                        return Err(());
+                    }
+                    Ok(())
+                });
+                tso_worker.spawn(wait_tso_fut);
```

Comment:
Could you avoid the nested spawn?

Reply:
It always need to spawn a future for collecting results.